src/arch/xtensa/lib/cpu.c

```diff
@@ -189,3 +189,28 @@ void cpu_power_down_core(void)
 	while (1)
 		arch_wait_for_interrupt(0);
 }
+
+int arch_cpu_restore_secondary_cores(void)
```

Comment:
This code will only restore cores which was active previously, such a mask will be used inside this function (it is not available globally)

Reply:
Why can we not have a global ```SHARED``` flag, possible as part of sof context ? I'm assuming this would be valid since it would be in the uncache alias.

Reply:
No cache issues here (only core0 will access it). That flag already exists - active_cores_mask, its static in cpu.c (but can be used anywhere from cpu_enabled_cores() if needed), 
It is used inside arch_cpu_restore_secondary_cores() (in arch_cpu_is_core_enabled(id) checks).

We can take it from cpu.c as cpu_enabled_cores(), run cpu_restore_secondary_cores(cpu_enabled_cores()) and inside anyway run helper functions like arch_cpu_is_core_enabled() which have local access to active cores mask. All of this is just weird and unnecessary. Maybe the name is just misleading and it will be better to call it cpu_restore_active_secondary_cores or something?